contracts/test/unit/VoltV2.t.sol

```diff
@@ -0,0 +1,384 @@
+        volt.approve(address(this), type(uint256).max);
+        assertEq(volt.allowance(from, address(this)), type(uint256).max);
+
+        volt.burnFrom(from, voltToBurn);
+
+        assertEq(volt.balanceOf(from), 0);
+        assertEq(volt.allowance(from, address(this)), type(uint256).max);
+        assertEq(volt.totalSupply(), 0);
+    }
+
+    function testBurnFromFailInsufficientBalance() public {
+        address from = address(0xFFF);
+        vm.prank(addresses.minterAddress);
+        volt.mint(from, 1e18);
+
+        vm.prank(from);
+        volt.approve(address(this), 2e18);
+
+        vm.expectRevert("Volt: burn amount exceeds balance");
+        volt.burnFrom(from, 2e18);
+    }
+
+    function testBurnFromFailInsufficientAllowance() public {
+        address from = address(0xFFF);
+        vm.prank(addresses.minterAddress);
+        volt.mint(from, 1e18);
+
+        vm.prank(from);
+        volt.approve(address(this), 0.9e18);
+
+        vm.expectRevert(stdError.arithmeticError);
+        volt.burnFrom(from, 1e18);
+    }
+
+    function testApprove(uint224 voltToApprove) public {
+        assertTrue(volt.approve(address(0xFFF), voltToApprove));
+        assertEq(volt.allowance(address(this), address(0xFFF)), voltToApprove);
+    }
+
+    function testTransfer(uint224 voltToTransfer) public {
+        vm.prank(addresses.minterAddress);
+        volt.mint(address(this), voltToTransfer);
+
+        volt.transfer(address(0xFFF), voltToTransfer);
+
+        assertEq(volt.totalSupply(), voltToTransfer);
+        assertEq(volt.balanceOf(address(this)), 0);
+        assertEq(volt.balanceOf(address(0xFFF)), voltToTransfer);
+    }
+
+    function testTransferFailInsufficientBalance() public {
+        vm.expectRevert(stdError.arithmeticError);
+        volt.transfer(address(0xFFF), 1e18);
+    }
+
+    function testTransferFailToVoltContract() public {
+        vm.expectRevert("Volt: cannot transfer to the volt contract");
+        volt.transfer(address(volt), 1e18);
+    }
+
+    function testTransferFrom(uint224 voltToTransfer) public {
+        vm.assume(voltToTransfer < type(uint224).max); // to make sure we don't run into infinite approval counter example
+        address from = address(0xFFF);
+        vm.prank(addresses.minterAddress);
+        volt.mint(from, voltToTransfer);
+
+        vm.prank(from);
+        volt.approve(address(this), voltToTransfer);
+        assertEq(volt.allowance(from, address(this)), voltToTransfer);
+
+        volt.transferFrom(from, address(this), voltToTransfer);
+
+        assertEq(volt.balanceOf(from), 0);
+        assertEq(volt.balanceOf(address(this)), voltToTransfer);
+        assertEq(volt.allowance(from, address(this)), 0);
+        assertEq(volt.totalSupply(), voltToTransfer);
+    }
+
+    function testTransferFromInfiniteApproval(uint224 voltToTransfer) public {
+        address from = address(0xFFF);
+        vm.prank(addresses.minterAddress);
+        volt.mint(from, voltToTransfer);
```

Comment:
we should make a test fixture that asserts that totalSupply and userBalance goes up anytime a mint occurs, and has a param of who to mint to. that way, assertions about the token happen automatically and we don't have to repeat this call to mint and prank code everywhere.